pkg/controller/gitopsconfig/controller.go

```diff
@@ -389,79 +392,111 @@ func (r *Reconciler) manageDeletion(instance *gitopsv1alpha1.GitOpsConfig) (reco
+	for _, j := range jobs {
+		if j.Labels != nil && j.Labels["action"] == "delete" {
+			deleters = append(deleters, j)
 		}
-		//we return because we need to wait for the job to stop
-		return reconcile.Result{
-			Requeue:      true,
-			RequeueAfter: time.Second * 5,
-		}, nil
-	}
-	//There should be only one pending job
-	job := applicableJobList[0]
-	if job.Status.Succeeded > 0 {
-		instance.ObjectMeta.Finalizers = removeString(instance.ObjectMeta.Finalizers, tagFinalizer)
-		if err := r.client.Update(context.TODO(), instance); err != nil {
-			log.Error(err, "unable to create update instance to remove finalizers", "instanceName", instance.Name)
-			return reconcile.Result{}, xerrors.Errorf("unable to create update instance %q to remove finalizers: %w", instance.Name, err)
+	}
+	if len(deleters) > 0 {
+		if len(deleters) > 1 {
+			log.Error(nil, "too many delete jobs found (expected 1)", "n", len(deleters), "instance", instance.Name)
+			// TODO: should we return here, or try to still do something sensible for user?
 		}
-		return reconcile.Result{}, nil
+		done := deleters[0].Status.Succeeded > 0
+		if !done {
+			// if it's not succeeded we wait for 5 seconds
+			// TODO add logic to stop at a certain point ... or not ...
+			// TODO add exponential backoff, possibly like in CronJob
+			return reconcile.Result{RequeueAfter: 5 * time.Second}, nil
+		}
+		return r.removeFinalizer(context.TODO(), instance)
 	}
-	//if it's not succeeded we wait for 5 seconds
-	//TODO add logic to stop at a certain point ... or not ...
-	//TODO add exponential backoff, possibly like in CronJob
+
+	log.Info("Launching delete job for instance", "instance", instance.Name)
+	_, err = r.createJob("delete", instance)
+	if err != nil {
+		log.Error(err, "unable to create deletion job", "instance", instance.Name)
+		return reconcile.Result{}, err
+	}
+	//we return because we need to wait for the job to stop
 	return reconcile.Result{
 		Requeue:      true,
 		RequeueAfter: time.Second * 5,
 	}, nil
+
+}
+
+func (r *Reconciler) removeFinalizer(ctx context.Context, instance *gitopsv1alpha1.GitOpsConfig) (reconcile.Result, error) {
+	instance.Finalizers = removeString(instance.Finalizers, tagFinalizer)
+	err := r.client.Update(ctx, instance)
+	if err != nil {
+		// if errors.IsConflict, then requeue
+		var errAPI errors.APIStatus
+		if xerrors.As(err, &errAPI) && errAPI.Status().Reason == metav1.StatusReasonConflict {
+			log.Error(err, "GitOpsConfig finalizer unable to remove itself; will retry", "instance", instance.Name)
+			return reconcile.Result{
+				RequeueAfter: 5 * time.Second,
+			}, xerrors.Errorf("GitOpsConfig finalizer unable to remove itself from %q, will retry: %w", instance.Name, err)
+		}
+		log.Error(err, "GitOpsConfig finalizer unable to remove itself", "instance", instance.Name)
+		return reconcile.Result{}, xerrors.Errorf("GitOpsConfig finalizer unable to remove itself from %q: %w", instance.Name, err)
+	}
+	return reconcile.Result{}, nil
```

Comment:
[optional] Do we want to log that finalizer was successfully deleted?

Reply:
Good idea, thanks!